### ISSUE TITLE
Assertion Error in normalized_coordinates

### ISSUE THREAD

**Issue by saraedum** [flatsurf/sage-flatsurf#87]
There's an unexpected assertion error reported by David Aulicino:

```
from flatsurf import EquiangularPolygons, similarity_surfaces
E = EquiangularPolygons((9, 7, 4, 2, 1, 1))
P = E((1, 1, 1, 5), normalized=True)
B = similarity_surfaces.billiard(P, rational=True)
S_mp = B.minimal_cover(cover_type="translation")
S = S_mp.erase_marked_points()
S, _ = S.normalized_coordinates()
```
which raises
```
AssertionError                            Traceback (most recent call last)
<ipython-input-18-0d15dd67650f> in <module>
      3 P = E((Integer(1), Integer(1), Integer(1), Integer(5)), normalized=True)
      4 print('P')
----> 5 B = similarity_surfaces.billiard(P, rational=True)
      6 print('B')
      7 S_mp = B.minimal_cover(cover_type="translation")

~/mambaforge/envs/flatsurf/lib/python3.8/site-packages/flatsurf/geometry/similarity_surface_generators.py in billiard(P, rational)
    509             base_ring = P.base_ring()
    510             C = ConvexPolygons(base_ring)
--> 511             comb_edges = P.triangulation()
    512             vertices = P.vertices()
    513             comb_triangles = build_faces(len(vertices), comb_edges)

~/mambaforge/envs/flatsurf/lib/python3.8/site-packages/flatsurf/geometry/polygon.py in triangulation(self)
    789         if len(self._v) == 3:
    790             return []
--> 791         return triangulate(self._v)
    792
    793     def translate(self, u):

~/mambaforge/envs/flatsurf/lib/python3.8/site-packages/flatsurf/geometry/polygon.py in triangulate(vertices)
    475                 if k == (i - 1) % n or k == i or \
    476                    k == (j - 1) % n or k == j:
--> 477                        assert segment_intersect(e, f) == 1
    478                        continue
    479                 res = segment_intersect(e, f)

AssertionError:
```
The problem ultimately comes from `triangulate` with the following input
```
from flatsurf.geometry.polygon import triangulate
x = polygen(QQ)
K.<c> = NumberField(x^2 - 3, embedding=AA(3).sqrt())
pts = [(0, 0), (1, 0), (1/2*c + 1, -1/2), (c + 1, 0), (-3/2*c + 1, 5/2), (0, c - 2)]
pts = [vector(K, v) for v in pts]
triangulate(pts)
```

**Comment by videlec** [comment]
The polygon is looking good

![polygon](https://user-images.githubusercontent.com/5619446/121798215-624cf900-cc25-11eb-9dd9-8f866095d955.png)

**Comment by videlec** [comment]
Fixed in #90
